Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -324,4 +331,127 @@ ExceptionOr<void> URLPattern::compileAllComponents(ScriptExecutionContext& conte
     return { };
 }
 
+static inline void matchHelperAssignInputsFromURL(const URL& input, String& protocol, String& username, String& password, String& hostname, String& port, String& pathname, String& search, String& hash)
+{
+    protocol = input.protocol().toString();
+    username = input.user();
+    password = input.password();
+    hostname = !input.host().isNull() ? input.host().toString() : emptyString();
+    port = input.port() ? String::number(*input.port()) : emptyString();
+    pathname = !input.path().isNull() ? input.path().toString() : emptyString();
+    search = !input.query().isNull() ? input.query().toString() : emptyString();
+    hash = !input.fragmentIdentifier().isNull() ? input.fragmentIdentifier().toString() : emptyString();
+}
+
+static inline void matchHelperAssignInputsFromInit(const URLPatternInit& input, String& protocol, String& username, String& password, String& hostname, String& port, String& pathname, String& search, String& hash)
+{
+    protocol = input.protocol;
+    username = input.username;
+    password = input.password;
+    hostname = input.hostname;
+    port = input.port;
+    pathname = input.pathname;
+    search = input.search;
+    hash = input.hash;
+}
+
+// https://urlpattern.spec.whatwg.org/#url-pattern-match
+ExceptionOr<std::optional<URLPatternResult>> URLPattern::match(ScriptExecutionContext& context, std::variant<URL, URLPatternInput>&& input, std::optional<String>&& baseURLString) const
+{
+    URLPatternResult result;
+    String protocol, username, password, hostname, port, pathname, search, hash;
+
+    if (URL* inputURL = std::get_if<URL>(&input)) {
+        ASSERT(!inputURL->isEmpty() && inputURL->isValid());
+        matchHelperAssignInputsFromURL(*inputURL, protocol, username, password, hostname, port, pathname, search, hash);
+        String urlAsString = inputURL->string();
+        result.inputs = Vector<URLPatternInput> { WTFMove(urlAsString) };
+    } else {
+        enum InputSwitchReturnTypes { ReturnTypeVoid, ReturnTypeNull, ReturnTypeException };
```

Comment:
`InputSwitchReturnTypes` => `InputSwitchReturnType`, and you may not need this.
Just have a `ExceptionOr<bool> hasError = false;` and update it as needed in the `switchOn` statements, like:
```
If (hasError.hasException())
    return inputCheckResult.releaseException();
if (hasError.returnValue())
    return { std::nullopt };
```